src/screens/Home/ActionButtons.js

```diff
@@ -89,258 +75,116 @@ const ActionButtonsWrapper = styled.View`
-            ...additionalInfo,
-            onPress: () => this.navigateToAction(acc, exchangeFlow),
-            label: `From ${additionalInfo.title}`,
-          };
-        });
-      default:
-        return [];
-    }
-  };
-
-  navigateToAction = (acc: Account, navigateTo: string, params?: Object) => {
-    const {
-      navigation,
-      activeWallet,
-      changeWalletAction,
-      blockchainNetwork,
-      setActiveBlockchainNetwork,
-    } = this.props;
-    const { type: walletType } = acc;
-    const { type: activeAccType } = activeWallet;
-
-    switch (walletType) {
-      case ACCOUNT_TYPES.SMART_WALLET:
-        if (blockchainNetwork !== BLOCKCHAIN_NETWORK_TYPES.ETHEREUM) {
-          setActiveBlockchainNetwork(BLOCKCHAIN_NETWORK_TYPES.ETHEREUM);
-        }
-        if (activeAccType !== ACCOUNT_TYPES.SMART_WALLET) {
-          changeWalletAction(acc, () => navigation.navigate(navigateTo, params));
-        } else {
-          navigation.navigate(navigateTo, params);
-        }
-        break;
-
-      case ACCOUNT_TYPES.KEY_BASED:
-        if (blockchainNetwork !== BLOCKCHAIN_NETWORK_TYPES.ETHEREUM) {
-          setActiveBlockchainNetwork(BLOCKCHAIN_NETWORK_TYPES.ETHEREUM);
-        }
-        if (activeAccType !== ACCOUNT_TYPES.KEY_BASED) {
-          changeWalletAction(acc, () => navigation.navigate(navigateTo, params));
-        } else {
-          navigation.navigate(navigateTo, params);
-        }
-        break;
-
-      default:
-        break;
+const ActionButtons = ({
+  navigation,
+  activeAccountAddress,
+  activeAccountBalances,
+  baseFiatCurrency,
+  rewardActive,
+  goToInvitationFlow,
+  rates,
+}: Props) => {
+  const [receiveAddress, setReceiveAddress] = useState('');
+  const [visibleAddFundsModal, setVisibleAddFundsModal] = useState(false);
+
+  const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+  const isSendButtonActive = calculateBalanceInFiat(rates, activeAccountBalances, fiatCurrency);
+
+  const addFundsModalOptions = [
+    {
+      key: 'buy',
+      label: `Buy with a card${Platform.OS === 'ios' ? ' or Apple Pay' : ''}`,
+      iconName: 'wallet',
+      onPress: () => navigation.navigate(EXCHANGE, { fromAssetCode: fiatCurrency }),
+    },
+    {
+      key: 'receive',
+      label: 'Send from another wallet',
+      iconName: 'qrDetailed',
+      onPress: () => setReceiveAddress(activeAccountAddress),
+    },
+    {
+      key: 'exchange',
+      label: 'Exchange',
+      iconName: 'flip',
+      onPress: () => navigation.navigate(EXCHANGE),
+    },
+    {
+      key: 'invite',
+      label: 'Invite and earn free tokens',
+      iconName: 'present',
+      hide: !rewardActive,
+      onPress: goToInvitationFlow,
+    },
+  ];
+
+  const closeAddFundsModal = (callback?: () => void) => {
+    setVisibleAddFundsModal(false);
+    if (callback) {
+      const timer = setTimeout(() => {
+        callback();
+        clearTimeout(timer);
+      }, 500);
```

Comment:
Honestly – I don't know, tried to lookup for it, but as I only cleanup up things and didn't find any relation so just left it here for now.

Reply:
I think this was done to give the modal some time to close, so we could display a toast after. Otherwise, the Toast will be mounted on closing modal and won't be displayed